Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp

```diff
@@ -43,11 +43,38 @@ namespace Platform
         }
         return physicalDeviceList;
     }
+    
+    float GetRefreshRate()
+    {
+        CGDirectDisplayID display = CGMainDisplayID();
+        CGDisplayModeRef currentMode = CGDisplayCopyDisplayMode(display);
+        return CGDisplayModeGetRefreshRate(currentMode);
+    }
 
-    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval)
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
     {
-        AZ_UNUSED(syncInterval);
-        [mtlCommandBuffer presentDrawable:drawable];
+        bool hasPresentAfterMinimumDurationAPI = false;
+        bool framePresented = false;
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
+        
+#if defined(__MAC_11_0)
```

Comment:
This will only be supported on Mac? not ios/tvos?

Reply:
yes. This code is only for Mac. iOS code is in the fil beloow and O3de does not support tvos.